### ISSUE TITLE
Mode subscriptions need some work

### ISSUE THREAD

**Issue by erodewald** [SmartThingsCommunity/smartapp-sdk-nodejs#45]
**Is your feature request related to a problem? Please describe.**
Subscribing to mode changes is possible, that's great, except proper usage is unclear. There's no `subscriptionName` for modes. 

In [smart-app.js#L177-178](https://github.com/SmartThingsCommunity/smartapp-sdk-nodejs/blob/master/lib/smart-app.js#L177-L178), it is clear that we can subscribe by name. 

It is evident in [smart-app.js#L411-L415](https://github.com/SmartThingsCommunity/smartapp-sdk-nodejs/blob/master/lib/smart-app.js#L411-L415) that the name is simply overridden.

See example below: (this works, but it's confusing)
```javascript
 .updated((context, updateData) => {
    context.api.subscriptions.unsubscribeAll().then(() => {
      context.api.subscriptions.subscribeToModeChange()
    })
  })
  // "modeChangeHandler" is basically discarded, so how is this handler functioning...?
  .subscribedEventHandler('modeChangeHandlerDoesntMatterWhatThisIsCalled', (context, event) => {
    const shouldTrigger = context.config.modes.some(cfg => {
      return cfg.modeConfig.modeId === event.modeId
    })
    console.log('ShouldTrigger:', shouldTrigger)
})
```

**Describe the solution you'd like**
Maybe there should be a unique function instead of re-using `subscribedEventHandler()`.

**Describe alternatives you've considered**
Ask responsible service to allow naming subscriptions to modes (and others) for our own sanity.

**Additional context**
n/a

**Comment by bflorian** [comment]
@erodewald I believe that this issue is addressed in #99 is it not? We still don't have support for returning handler names in these events from the platform, but we do now have [unique function names](https://github.com/SmartThingsCommunity/smartapp-sdk-nodejs/blob/master/lib/smart-app.js#L644-L648) for these handlers

**Comment by bflorian** [comment]
Closing since SDK is doing the best it can given that the platform doesn't support multiple named handlers for these types of events.
